### PR TITLE
Spring security adding

### PR DESCRIPTION
**Pull request by Damrocka89**
- added spring security, h2 db to pom
- added users package including model, controller, dto for registration, user repository, user and security services, user validator, spring security config
- added basic templates for login and registration
- added application properties for h2 and hibernate
- added messages.properties including messages for registration errors descriptions

### CODE REVIEW

File: src/main/java/pl/sda/giftwisher/giftwisher/users/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import pl.sda.giftwisher.giftwisher.users.dto.RegistrationFormDTO;
+import pl.sda.giftwisher.giftwisher.users.service.SecurityService;
+import pl.sda.giftwisher.giftwisher.users.service.UserService;
+import pl.sda.giftwisher.giftwisher.users.validator.UserValidator;
+
+@Controller
+public class UserController {
+
+    private UserService userService;
```
Comment (towczare) [inline src/main/java/pl/sda/giftwisher/giftwisher/users/controller/UserController.java]: Dobrą praktyką jest tworzenie takich serwisów jako final. Czyli byłoby:
`
    private final UserService userService;
    private final SecurityService securityService;
    private final UserValidator userValidator;
`

File: src/main/java/pl/sda/giftwisher/giftwisher/users/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    @PostMapping("/registration")
+    public String registration(@ModelAttribute("userForm") RegistrationFormDTO userForm, BindingResult bindingResult) {
+        userValidator.validate(userForm, bindingResult);
+
+        if (bindingResult.hasErrors()) {
+            return "registration";
+        }
+
+        userService.save(userForm);
+
```
Comment (towczare) [inline src/main/java/pl/sda/giftwisher/giftwisher/users/controller/UserController.java]: Te spacje można pousuwać, rozwlekają kod :-)

Comment (Damrocka89) [reply src/main/java/pl/sda/giftwisher/giftwisher/users/controller/UserController.java]: usune

File: src/main/java/pl/sda/giftwisher/giftwisher/users/controller/UserController.java (hunk 1 above)
Comment (Damrocka89) [reply src/main/java/pl/sda/giftwisher/giftwisher/users/controller/UserController.java]: ok

**Comment by Damrocka89** [comment]
refactoring

**Review by dpierz — APPROVED** [review]

**Pull request merged**